docs/user/kotlin-impl-spec-comparison.md

```diff
@@ -0,0 +1,62 @@
+the [PartiQL specification](https://partiql.org/assets/PartiQL-Specification.pdf).
+Also included are some features that are part of the Kotlin implementation but are not yet formally defined in the 
+PartiQL spec.
+
+Note: the list is only partially completed. Sections 6.2 onwards from the spec have yet to be compared with the Kotlin
+implementation.
+
+## Features defined in spec
+
+| Spec section	              |Feature	|Implemented in `partiql-lang-kotlin`	| Notes	                                                                                                                                                                                                             |
+|----------------------------|---	|---	|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| 3.3	                       |Syntax improvement `SELECT` optionally written as last clause	|No	| 	                                                                                                                                                                                                                  |
+| 4	                         |Tuple path navigation	|Yes	| `t.a`; tuple `t` and attribute `a` (identifier)	                                                                                                                                                                   |
+| 4	                         |Array navigation	|Yes	| `a[i]`; array `a`, `i` is an expression that evaluates to an integer	                                                                                                                                              |
+| 4	                         |Tuple navigation with array notation	|Yes	| `a[s]`; shorthand for `a.s` where `a` is a tuple, `s` is either a string literal or expression CAST to a string	                                                                                                   |
+| 4	                         |Composition of naviagations	|Yes	| `r.no[1]`	                                                                                                                                                                                                         |
+| 4	                         |Tuple naviagation in tuples w/ duplicate attributes	|Yes	| 	                                                                                                                                                                                                                  |
+| 4.1	                       |Tuple path evaluation on wrongly typed data	|Partially^1	| Permissive mode behavior returns `MISSING` as defined in the spec. "STRICT" type-checking mode doesn't exist in Kotlin implementation	                                                                             |
```

Comment:
I understand we have a reference to "strict mode" in the Spec. but we also have references to "Type checking mode" which has been put against "Permissive mode". Which one are we denoting here by "STRICT"?

Reply:
I've been under the impression that "type checking mode" was the `STRICT` mode in the spec, so `STRICT` in this document refers to "type checking mode".

Reply:
Yeah, I think using "type checking mode" rather than `STRICT` is more clear and representative of the spec. I'll change in the next commit.